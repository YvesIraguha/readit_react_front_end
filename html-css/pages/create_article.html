<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
    <link rel="stylesheet" href="../css/main.css" />
    <title>READ_IT</title>
  </head>
  <body>
    <section class="navbar">
      <div class="logo">
        <img src="../icons/open-book.svg" alt="Logo" class="icon" />
      </div>
      <div class="profile">
        <div class="avatar">
          <img src="../images/user.png" alt="Profile" class="icon" />
        </div>
        <div class="username">
          <p>Joe Doe</p>
        </div>
      </div>
    </section>
    <div class="create_article">
      <div class="title">
        <p>Add your title</p>
        <hr />
      </div>
      <div>
        <p>Tell story</p>
      </div>
      <button class="btn">Save</button>
    </div>
  </body>
</html>
