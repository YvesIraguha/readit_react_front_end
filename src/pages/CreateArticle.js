import React, { Component } from 'react';
import RichTextEditor from 'react-rte';
import { createArticle } from '../redux/actions/createArticleAction';
import Input from '../components/common/Input';
import { connect } from 'react-redux';

class CreateArticle extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: RichTextEditor.createEmptyValue()
    };
  }

  onInputChange(event) {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }
  onEditorChange = value => {
    this.setState({ value });
  };

  submitArticle = () => {
    const { title, value } = this.state;
    const { createArticle } = this.props;
    createArticle({ title, content: value.toString('html') });
  };
  render() {
    return (
      <div>
        <div className="create_article">
          <div className="title">
            <Input
              type="title"
              name="title"
              placeholder="Title here..."
              onChange={e => this.onInputChange(e)}
            />
            <hr />
          </div>
          <div>
            <RichTextEditor
              value={this.state.value}
              onChange={e => this.onEditorChange(e)}
            />
          </div>
          <button className="btn" onClick={() => this.submitArticle()}>
            Save
          </button>
        </div>
      </div>
    );
  }
}

export default connect(
  null,
  { createArticle }
)(CreateArticle);
