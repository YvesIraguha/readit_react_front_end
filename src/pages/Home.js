import React, { Component } from "react";
import NavBar from "../components/common/NavBar";
import { fetchAllArticles } from "../redux/actions/createArticleAction";
import ArticleCard from "../components/common/ArticleCard";
import { connect } from "react-redux";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      articles: []
    };
  }
  componentDidMount() {
    const { fetchAllArticles } = this.props;
    fetchAllArticles();
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.articles !== this.state.articles) {
      this.setState({ articles: nextProps.articles });
    }
  }
  render() {
    const { articles } = this.state;
    return (
      <div>
        <NavBar />
        <div className="home_container">
          <div className="articles">
            {articles.map(article => (
              <ArticleCard key={article.id} {...article} />
            ))}
          </div>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  articles: state.article.articles
});
export default connect(
  mapStateToProps,
  { fetchAllArticles }
)(Home);
