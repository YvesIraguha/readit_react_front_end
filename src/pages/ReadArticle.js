import React, { Component } from 'react';
import NavBar from '../components/common/NavBar';
import {
  fetchingArticle,
  deleteArticle,
  updatedArticle
} from '../redux/actions/createArticleAction';
import { connect } from 'react-redux';
import ReactHtmlParser from 'react-html-parser';

class ReadArticle extends Component {
  constructor(props) {
    super(props);
    this.state = {
      slug: '',
      article: {
        title: '',
        content: ''
      }
    };
  }
  componentDidMount() {
    const {
      fetchingArticle,
      match: {
        params: { slug }
      }
    } = this.props;
    this.setState({ slug });
    fetchingArticle(slug);
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.article.article.post !== this.state.article) {
      this.setState({ article: { ...nextProps.article.article.post } });
    }
  }

  render() {
    const {
      slug,
      article: { title, content }
    } = this.state;
    const { deleteArticle } = this.props;
    return (
      <div>
        <NavBar />
        <div className="article">
          <div className="article_text">
            <h1>{title}</h1>
            <div>{ReactHtmlParser(content)}</div>
            <button onClick={() => deleteArticle(slug)}>Delete</button>
          </div>
          <div className="article_comments">
            <div className="new_comment">
              <div className="avatar">
                <img
                  src={require('../assets/images/user.png')}
                  alt="User"
                  className="icon"
                />
                <p>Joe Doe</p>
              </div>
              <div>
                <p>Add comment here</p>
                <button className="btn">Save</button>
              </div>
              <hr />
            </div>
            <div className="comment">
              <div className="avatar">
                <img
                  src={require('../assets/images/user.png')}
                  alt="user"
                  className="icon"
                />
                <p>Joe Doe</p>
              </div>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                eiusmod tempor incididunt ut labore et dolore magna aliqua.
                Lorem ipsum dolor sit amet,
              </p>
              <hr />
            </div>
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  article: state.article
});
export default connect(
  mapStateToProps,
  { fetchingArticle, deleteArticle }
)(ReadArticle);
