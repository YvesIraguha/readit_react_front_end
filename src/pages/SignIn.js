import React, { Component } from 'react';
import { logIn } from '../redux/actions/loginActions';
import Input from '../components/common/Input';
import { connect } from 'react-redux';
const mapDispatchToProps = dispatch => ({
  signIn: user => dispatch(logIn(user))
});
class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  onInputChange(event) {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }
  submitLoginCredentials() {
    const { email, password } = this.state;
    const { signIn } = this.props;
    signIn({ email, password });
  }
  render() {
    return (
      <div>
        <div className="container">
          <div className="card">
            <div className="email sign_up">
              <img
                src={require('../assets/icons/email.svg')}
                alt="Email"
                className="icon"
              />
              <Input
                type="email"
                name="email"
                placeholder="Email ..."
                onChange={e => this.onInputChange(e)}
              />
            </div>
            <div className="password sign_up">
              <img
                src={require('../assets/icons/locked.svg')}
                alt="password"
                className="icon"
              />
              <Input
                type="password"
                name="password"
                placeholder="Password ..."
                onChange={e => this.onInputChange(e)}
              />
            </div>
            <button
              className="btn"
              onClick={() => this.submitLoginCredentials()}
            >
              Sign In
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default connect(
  null,
  mapDispatchToProps
)(SignIn);
