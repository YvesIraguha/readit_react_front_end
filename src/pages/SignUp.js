import React, { Component } from 'react';
import Input from '../components/common/Input';
import { signUp } from '../redux/actions/singupAction';
import { connect } from 'react-redux';
export const mapDispatchToProps = dispatch => ({
  sendSignUpInfo: user => dispatch(signUp(user))
});
class SignUp extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  onInputChange(event) {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  submitSignUpCredentials() {
    const { email, password, lastName, firstName } = this.state;
    const { sendSignUpInfo } = this.props;
    sendSignUpInfo({ email, password, lastName, firstName });
  }
  render() {
    return (
      <div>
        <div className="container">
          <div className="card">
            <div className="email sign_up">
              <img
                src={require('../assets/icons/email.svg')}
                alt="Email"
                className="icon"
              />
              <Input
                type="email"
                name="email"
                placeholder="Email ..."
                onChange={e => this.onInputChange(e)}
              />
              <Input
                type="email"
                name="lastName"
                placeholder="Last name ..."
                onChange={e => this.onInputChange(e)}
              />
              <Input
                type="email"
                name="firstName"
                placeholder="first name ..."
                onChange={e => this.onInputChange(e)}
              />
            </div>
            <div className="password sign_up">
              <img
                src={require('../assets/icons/locked.svg')}
                alt="password"
                className="icon"
              />
              <Input
                type="password"
                name="password"
                placeholder="Password ..."
                onChange={e => this.onInputChange(e)}
              />
            </div>
            <button
              className="btn"
              onClick={() => this.submitSignUpCredentials()}
            >
              Sign Up
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default connect(
  null,
  mapDispatchToProps
)(SignUp);
