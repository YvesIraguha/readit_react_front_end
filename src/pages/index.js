import React from 'react';
import { connect } from 'react-redux';
import { trialMethod } from '../redux/actions/index';
export const HelloWorld = () => <h1>Hello there world!</h1>;
const Button = props => (
  <button className="btn" onClick={e => props.firstMethod()}>
    Click!
  </button>
);
const mapDispatchToProps = dispatch => ({
  firstMethod: () => dispatch(trialMethod())
});

export default connect(
  null,
  mapDispatchToProps
)(Button);
