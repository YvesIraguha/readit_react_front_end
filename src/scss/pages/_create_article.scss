.create_article {
  width: 60%;
  margin: auto;
}

@media (min-width: 768px) {
  .create_article {
    width: 60%;
  }
}

@media (max-width: 767px) {
  .create_article {
    width: 80%;
    font-size: 18px;
  }
}
