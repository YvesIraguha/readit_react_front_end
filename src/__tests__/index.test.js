import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { HelloWorld } from '../pages/index';
configure({ adapter: new Adapter() });
test('should return true', () => {
  expect(true).toBe(true);
});

test('should match the snapshot', () => {
  const wrapper = shallow(<HelloWorld />);
  expect(wrapper).toMatchSnapshot();
});
