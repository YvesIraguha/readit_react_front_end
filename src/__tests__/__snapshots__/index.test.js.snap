// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<h1>
  Hello there world!
</h1>
`;
